### PR TITLE
Copy over changes from other repo (mostly a continuation of validator fixes)

### LINKED ISSUE #144: Additional `MSG_` macros
While going and fixing validator warnings for #7 I'm noticing that there are a lot of repeated message captions for certain characters who speak a lot but don't have a dedicated macro for them in `macros/messages.cfg`, for example Hoyre and Huon Ryedric

### PR DESCRIPTION
**Pull request by cooljeanius**
This PR contains the following:
- new linting workflow
- recompression of images
- start of work on some lua code
- validator fixes
- updated README

See the following issues:
- partially inspired by issue #7
- partially addresses a follow-up comment I made in issue #32
- lets us check off checkbox 2 in issue #33
- using the validator had a bit of the same effect that attempting to address issue #57 would have had
- gets started with issue #73 (only a bit, though; I'm holding back on some other parts I've written for it until they're more submittable)
- adds some FIXMEs before lines that the validator pointed out that I think might be causing issue #81
- Closes #144 (at least I think that ought to be enough for now)

### CODE REVIEW

File: macros/ano_macros.cfg [hunk 1]
```diff
@@ -28,12 +28,7 @@
     [/filter]
 #enddef
 
-#define SCROLL_TO X Y
-    [scroll_to]
-        x={X}
-        y={Y}
-    [/scroll_to]
-#enddef
+# (SCROLL_TO is a mainline macro; no need to define it ourselves)
```
Comment (nemaara) [inline macros/ano_macros.cfg:31]: Just delete, no need for comment.

**Review by nemaara — COMMENTED** [review]
I assume you playtested these changes? As in nothing breaks when you load the scenarios.

**Comment by cooljeanius** [comment]
> I assume you playtested these changes? As in nothing breaks when you load the scenarios.

I only tested the first few scenarios, since the `--validate-addon` flag will check the entire campaign upon just loading the first scenario... trying to switch to some other scenarios with the `:cl` in-game debug command, though, reveals a bug that was probably already there previously that I'd never noticed before due to not having tried switching levels this way:
when skipping from S04 ("Battle of Barnon") to S05 ("The Swamp Things") I get:
```
20240103 01:29:55 info wml: A [recall] tag with the following content failed:
show = no
side = 1
x = 1
y = 1
20240103 01:29:57 error wml: [unstore_unit]: variable 'ano_reme_stored' doesn't contain unit data
```
I'll open a separate issue about that. Note that I'm planning several more PRs and another full playthrough before the next release, so if anything else comes up, I can fix it later...

**Comment by cooljeanius** [comment]
> I'll open a separate issue about that. 

...ok, I opened #165 for it.

File: macros/ano_macros.cfg (hunk 1 above)
Comment (cooljeanius) [reply macros/ano_macros.cfg]: ok, removed in [a445cf8](https://github.com/nemaara/A_New_Order/pull/163/commits/a445cf8f1e72dd6801aa9756cd932b647ecb4170).

**Pull request merged**